### PR TITLE
Assorted bug fixes and minor enhancements

### PR DESCRIPTION
**Pull request by simonziegler**

### CODE REVIEW

File: Material.Blazor/Components/DatePicker/InternalDatePickerDayButton.razor [hunk 1]
```diff
@@ -13,4 +13,5 @@
           class="@classes"
           Disabled="@ButtonDisabled"
           Label="@Day"
-          @onclick="@OnClickAsync" />
\ No newline at end of file
+          @onclick="@OnClickAsync"
+          type="button" />
```
Comment (stefanloerwald) [inline Material.Blazor/Components/DatePicker/InternalDatePickerDayButton.razor:17]: Out of curiosity: what's the issue when we don't include `type="button"`?

File: Material.Blazor/Foundation/BatchingJsRuntime.cs [hunk 2]
```diff
@@ -74,27 +85,19 @@ private async void Timer_Elapsed(object sender, ElapsedEventArgs e)
...
-        /// </summary>
-        /// <typeparam name="T"></typeparam>
-        /// <param name="identifier"></param>
-        /// <param name="args"></param>
-        /// <returns></returns>
-        public async Task<T> InvokeAsync<T>(string identifier, params object[] args)
+
+
+        /// <inheritdoc/>
+        public async ValueTask<T> InvokeAsync<T>(string identifier, params object[] args)
```
Comment (stefanloerwald) [inline Material.Blazor/Foundation/BatchingJsRuntime.cs]: I'm contemplating a PR that would also support batching for `InvokeAsync<T>`, but that requires a `Task<T>` return type. Any reason this must be a `ValueTask<T>`?

File: Material.Blazor/Foundation/IBatchingJsRuntime.cs [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
 {
-    public interface IBatchingJsRuntime
+    /// <summary>
+    /// A service that wraps IJSRuntime, batching calls to InvokeVoidAsync().
+    /// </summary>
+    public interface IBatchingJSRuntime
     {
-        Task<T> InvokeAsync<T>(string identifier, params object[] args);
+        /// <summary>
+        /// Same as <see cref="JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime, string, object[])"/>, except calls are batched in 20ms intervals.
```
Comment (stefanloerwald) [inline Material.Blazor/Foundation/IBatchingJsRuntime.cs]: Classic case of docs are out of sync with code. The interval is actually 10ms at the moment. My bad.

File: Material.Blazor/Components/BladeSet/MBBladeComponent.cs [hunk 4]
```diff
@@ -27,9 +27,19 @@ public abstract class MBBladeComponent<TParam> : ComponentBase where TParam : MB
...
+
+
         /// <summary>
         /// A utility function to close the blade, calling BladeSet.RemoveBladeAsync(), passing the blade reference.
         /// </summary>
-        protected async Task CloseBladeAsync() => await BladeSet.RemoveBladeAsync(BladeReference).ConfigureAwait(false);
+        public async Task CloseBladeAsync()
+        {
+            IsOpen = false;
+            await BladeSet.RemoveBladeAsync(BladeReference).ConfigureAwait(false);
```
Comment (stefanloerwald) [inline Material.Blazor/Components/BladeSet/MBBladeComponent.cs]: You could also do `return BladeSet.RemoveBladeAsync(BladeReference)` which leaves any `ConfigureAwait` config to the caller.

File: Material.Blazor/Components/Card/MBCard.razor.cs [hunk 5]
```diff
@@ -81,7 +111,7 @@ private protected override async Task DestroyMcwComponent()
         {
             if (PrimaryAction != null)
             {
-                await JsRuntime.InvokeVoidAsync("MaterialBlazor.MBCard.destroy", PrimaryActionReference);
+                //await JsRuntime.InvokeVoidAsync("MaterialBlazor.MBCard.destroy", PrimaryActionReference);
```
Comment (stefanloerwald) [inline Material.Blazor/Components/Card/MBCard.razor.cs]: We do not call `DestroyMcwComponent` anymore, right? Doesn't that make this change obsolete?

File: Material.Blazor/Foundation/ComponentFoundation.cs [hunk 6]
```diff
@@ -137,10 +137,10 @@ protected virtual async ValueTask DisposeAsync(bool disposing)
                 return;
             }
 
-            if (HasInstantiated)
-            {
-                await DestroyMcwComponent();
-            }
+            //if (HasInstantiated)
+            //{
+            //    await DestroyMcwComponent();
```
Comment (stefanloerwald) [inline Material.Blazor/Foundation/ComponentFoundation.cs]: We should consider marking `DestroyMcwComponent` as obsolete to make clean up easier

File: Material.Blazor/Components/DatePicker/InternalDatePickerDayButton.razor (hunk 1 above)
Comment (simonziegler) [reply Material.Blazor/Components/DatePicker/InternalDatePickerDayButton.razor:17]: Now I'm trying to remember. I *think* it was about having a DatePicker in a form. If you don'tuse `type="button"` as recall the form interprets a button click as a submit action.

File: Material.Blazor/Foundation/BatchingJsRuntime.cs (hunk 2 above)
Comment (simonziegler) [reply Material.Blazor/Foundation/BatchingJsRuntime.cs]: Not my strong point and I'm happy to follow your lead.

File: Material.Blazor/Components/BladeSet/MBBladeComponent.cs (hunk 4 above)
Comment (simonziegler) [reply Material.Blazor/Components/BladeSet/MBBladeComponent.cs]: Good idea. That change is being committed shortly. I'd like to have a general `ConfigureAwait()` call with you for my own education.

File: Material.Blazor/Components/Card/MBCard.razor.cs (hunk 5 above)
Comment (simonziegler) [reply Material.Blazor/Components/Card/MBCard.razor.cs]: Leaving that whole function in place was a bug and I've removed it.

File: Material.Blazor/Foundation/ComponentFoundation.cs (hunk 6 above)
Comment (simonziegler) [reply Material.Blazor/Foundation/ComponentFoundation.cs]: Good idea. Doing that now.

Comment (simonziegler) [reply Material.Blazor/Foundation/ComponentFoundation.cs]: Better still I've just deleted all `DestroyMcwComponent`s from the C# code. I haven't yet looked at removing the JS code. I'm in two minds and am OK with leaving it around for a while.

File: Material.Blazor/Foundation/ComponentFoundation.cs [hunk 7]
```diff
@@ -123,12 +123,6 @@ public abstract class ComponentFoundation : ComponentBase, IAsyncDisposable
...
 
 
-        /// <summary>
-        /// Components should override this with a function to be called when Material.Blazor wants to run Material Components Web instantiation via JS Interop - always gets called from <see cref="OnAfterRenderAsync()"/>, which should not be overridden.
-        /// </summary>
-        private protected virtual async Task DestroyMcwComponent() => await Task.CompletedTask;
-
-
         private bool _disposed;
         protected virtual async ValueTask DisposeAsync(bool disposing)
```
Comment (stefanloerwald) [inline Material.Blazor/Foundation/ComponentFoundation.cs]: This method is now synchronous, which is a good thing! We should review all instances where we inherit from this and see whether we actually still need this to be async (IIRC this was only to do async JS calls, which are now removed anyway).

File: Material.Blazor/Components/Card/MBCard.scss [hunk 8]
```diff
@@ -15,15 +15,6 @@
...
 
-.mb-card__secondary, .mb-card__subtitle {
-    color: rgba(0,0,0,.54);
-    color: var(--mdc-theme-text-secondary-on-background,rgba(0,0,0,.54))
-}
-
-.mb-card__secondary {
-    padding: 0 1rem 8px
-}
-
```
Comment (stefanloerwald) [inline Material.Blazor/Components/Card/MBCard.scss]: This removal changes font colors

File: Material.Blazor/Foundation/BatchingJsRuntime.cs (hunk 2 above)
Comment (stefanloerwald) [reply Material.Blazor/Foundation/BatchingJsRuntime.cs]: Changed back to `Task<T>` in 9fd9b54

**Comment by stefanloerwald** [comment]
Skim-reading through the diff, it seems it boils down to

- additional `<div class="mdc-card__content">` wrapper and accordingly additional indentation (btw: this would not have been included in the diff if it weren't for the next change)
- renamed `mb-card__primary` to `mb-card__autostyled`
- additional `type="button"` on buttons in datepicker

Given that the visuals for cards remain correct, I think this is a go-ahead from my side.

**Review by stefanloerwald — APPROVED** [review]

**Comment by stefanloerwald** [comment]
Hold on a minute, I'll push some changes regarding unnecessary using directives

**Comment by stefanloerwald** [comment]
Done & ready to merge

**Pull request merged**